.github/workflows/make-release.yml

```diff
@@ -0,0 +1,30 @@
+# Copyright 2021-2024 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+name: "Make release"
+
+on:
+  push:
+    tags:
+      - "openssl-*"
+
+jobs:
+  release:
+    runs-on: "releaser"
+    steps:
+    - name: "Prepare assets"
+      run: |
+        curl -L -O -H "Authorization: Bearer ${{ secrets.GHE_TOKEN }}" https://github.openssl.org/openssl/openssl/archive/refs/tags/${{ github.ref_name }}.tar.gz
```

Comment:
Unfortunately, github uses a prefix directory named `{project}-{tag}`, i.e. your test produced this:

``` console
$ curl -L -O https://github.com/quarckster/openssl-fork/releases/download/openssl-3.3.2/openssl-3.3.2.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 17.4M  100 17.4M    0     0  8329k      0  0:00:02  0:00:02 --:--:-- 11.1M
$ tar -tvzf openssl-3.3.2.tar.gz |head
drwxrwxr-x root/root         0 2024-09-03 14:46 openssl-openssl-3.3.2/
drwxrwxr-x root/root         0 2024-09-03 14:46 openssl-openssl-3.3.2/.ctags.d/
-rw-rw-r-- root/root       416 2024-09-03 14:46 openssl-openssl-3.3.2/.ctags.d/add-dir.ctags
-rw-rw-r-- root/root       453 2024-09-03 14:46 openssl-openssl-3.3.2/.ctags.d/exclude.ctags
drwxrwxr-x root/root         0 2024-09-03 14:46 openssl-openssl-3.3.2/.ctags.d/openssl-stage1/
-rw-rw-r-- root/root       471 2024-09-03 14:46 openssl-openssl-3.3.2/.ctags.d/openssl-stage1/10extrac-macrodefs.ctags
drwxrwxr-x root/root         0 2024-09-03 14:46 openssl-openssl-3.3.2/.ctags.d/openssl-stage2/
-rw-rw-r-- root/root       354 2024-09-03 14:46 openssl-openssl-3.3.2/.ctags.d/openssl-stage2/10expand-macros.ctags
-rw-rw-r-- root/root       159 2024-09-03 14:46 openssl-openssl-3.3.2/ACKNOWLEDGEMENTS.md
-rw-rw-r-- root/root       990 2024-09-03 14:46 openssl-openssl-3.3.2/AUTHORS.md
```

[sigh] and there's nothing you can do to change that, 'cause github has decided...

So the options we have are:

- not to care, i.e. accept the new prefix directory (and take the shitstorm we're may get, 'cause users often care)
- find a way to massage the tarball

Reply:
@levitte I noticed that too. I'm not sure how the prefix is critical but if it is than it would be easier to use existing script

@t8m Unfortunately, GitHub does not support `git archive`.

Reply:
BTW, the prefix directory becomes even more horrible if we use this on pre-3.0 versions...

Reply:
> @t8m Unfortunately, GitHub does not support `git archive`.

So?  Instead of downloading github's tarball, we could do a checkout at the given tag and run `util/mktar.sh` in the checked out directory.  I'm guessing something like this:

``` yaml
  - uses: actions/checkout@v4
    with:
      repository: github.com/openssl/openssl
      ref: {{ github.ref_name }}
      path: openssl
  - name: Prepare OpenSSL tarball
    run: cd openssl && util/mktar.sh && mv *.tar.gz ..
```

Of course, you'll have to find the tarball (`*.tar.gz`) and use the name you find as basis for all other file names.

Reply:
Ok, let me update the workflow.